### ISSUE TITLE
[Accessibility]: Announce contextual information for repo sections in ref panel

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36578]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33510

### Problem description

When navigating the ref panel with a screen reader, it is hard to understand whether one should go into a repo section or not because there's not enough context. Does this repo have 10 files that matched? How many matches did this repo have in total compared to a different repo?

### Expected behavior

It would be nice to announce the number of files that matches and total number of matches.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
